drivers/flash/Kconfig

```diff
@@ -86,10 +86,6 @@ menuconfig FLASH
 
 if FLASH
 
-module = FLASH
```

Comment:
To make this change we have to deviate from the pattern present in other driver `Kconfig` files (I looked at the serial and SPI ones for example).

Could we instead disable logging at the SOC level, maybe in `soc/nxp/common/Kconfig.flexspi_xip`?

Reply:
no, I don't care about this pattern, it probably is done thoughtlessly anyways, specific drivers should be able to override driver class config defaults, the other classes just haven't run into the issue yet. I am not putting it in the SOC level because the default value we want has specifically to do with the driver, it belongs here

Reply:
I think other classes have run into this issue, and the solution was to define a Kconfig for that *driver's* log level, rather than using the subsystem Kconfig: https://github.com/zephyrproject-rtos/zephyr/blob/cfe5bd922432d9c2090e9cbf149edd7172dc4e62/drivers/gpio/Kconfig.fxl6408#L20 Perhaps we should do the same here, because I could see someone wanting debug logging to be enabled for their on chip flash driver, but disabled for the FlexSPI

Reply:
Looking at this more, some driver Kconfig files define the logging string later (such as I2C: https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/i2c/Kconfig#L143) and some do not. So I think this approach is fine as is